include/swift/AST/Types.h

```diff
@@ -1351,12 +1411,20 @@ class TupleTypeElt {
 
   /// Retrieve a copy of this tuple type element with the type replaced.
   TupleTypeElt getWithType(Type T) const {
-    return TupleTypeElt(T, getName(), isVararg());
+    return TupleTypeElt(T, getName(), isVararg(), isAutoClosure(),
+                        isEscaping());
   }
 
   /// Retrieve a copy of this tuple type element with the name replaced.
   TupleTypeElt getWithName(Identifier name = Identifier()) const {
-    return TupleTypeElt(getType(), name, isVararg());
+    return TupleTypeElt(getType(), name, isVararg(), isAutoClosure(),
+                        isEscaping());
+  }
+
+  /// Retrieve a copy of this tuple type element with the type and name
+  /// replaced.
+  TupleTypeElt getWithTypeAndName(Type T, Identifier name) const {
```

Comment:
I feel like it's fine to use constructor syntax for this one rather than a `getWith…` method. `TupleTypeElt(T, name, orig.getParameterFlags()`.